### PR TITLE
make abs op support complex types

### PR DESCRIPTION
**Pull request by MingMingShangTian**
### PR types

New features
### PR changes

OPs
### Describe

[Complex] support complex calc for abs op

### CODE REVIEW

File: paddle/fluid/operators/abs_op.cc [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+class AbsOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "(Tensor), The input tensor of abs op.");
+    AddOutput("Out", "(Tensor), The output tensor of abs op.");
+    AddComment(R"DOC(
+Abs Operator.
+
+This operator is used to perform elementwise abs for input $X$.
```
Comment (chenwhql) [inline paddle/fluid/operators/abs_op.cc:58]: Use abs calculation formula here. like
```
$$out = |x|$$
```

File: paddle/fluid/operators/abs_op.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/fluid/framework/eigen.h"
```
Comment (chenwhql) [inline paddle/fluid/operators/abs_op.h]: not need eigen.h?

File: paddle/fluid/operators/activation_op.cc [hunk 3]
```diff
@@ -714,7 +714,6 @@ REGISTER_ACTIVATION_OP_MAKER(Tanh, TanhDoc);
 REGISTER_ACTIVATION_OP_MAKER(TanhShrink, TanhShrinkDoc);
 REGISTER_ACTIVATION_OP_MAKER(Sqrt, SqrtDoc);
 REGISTER_ACTIVATION_OP_MAKER(Rsqrt, RsqrtDoc);
-REGISTER_ACTIVATION_OP_MAKER(Abs, AbsDoc);
```
Comment (chenwhql) [inline paddle/fluid/operators/activation_op.cc:717]: AbsDoc also need to be removed

File: paddle/fluid/operators/abs_op.cc (hunk 1 above)
Comment (MingMingShangTian) [reply paddle/fluid/operators/abs_op.cc:58]: Done

File: paddle/fluid/operators/abs_op.h (hunk 2 above)
Comment (MingMingShangTian) [reply paddle/fluid/operators/abs_op.h]: removed this include

File: paddle/fluid/operators/activation_op.cc (hunk 3 above)
Comment (MingMingShangTian) [reply paddle/fluid/operators/activation_op.cc:717]: removed

**Review by GaoWei8 — DISMISSED** [review]
LGTM for op benchmark

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by GaoWei8 — APPROVED** [review]
LGTM for op benchmark

**Review by chenwhql — APPROVED** [review]

**Pull request merged**